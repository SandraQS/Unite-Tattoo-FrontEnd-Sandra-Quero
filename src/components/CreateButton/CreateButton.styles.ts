import { StyleSheet } from "react-native";

export default StyleSheet.create({
  card: {
    borderRadius: 20,
    flex: 1,
    flexDirection: "column",
    margin: 5,
    padding: 10,
  },

  imageContainer: {
    alignContent: "center",
    borderRadius: 20,
    flex: 5,
    height: 200,
    justifyContent: "center",
  },

  mainContainerCard: {
    height: 300,
    width: 170,
  },
});
