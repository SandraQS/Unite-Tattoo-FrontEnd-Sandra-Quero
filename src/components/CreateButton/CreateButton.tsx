import React from "react";
import { TouchableOpacity, View } from "react-native";
import AutoHeightImage from "react-native-auto-height-image";
import { CreateButtonProps } from "../../types/components.types";

import style from "./CreateButton.styles";

const CreateButton = ({ functionCreate }: CreateButtonProps) => {
  return (
    <TouchableOpacity onPress={functionCreate} activeOpacity={0.6}>
      <View style={style.card}>
        <View style={style.imageContainer}>
          <AutoHeightImage
            width={150}
            source={require("../../assets/crear-rombo.png")}
          />
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default CreateButton;
