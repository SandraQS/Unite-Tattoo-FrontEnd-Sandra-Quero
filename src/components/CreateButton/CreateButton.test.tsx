import { fireEvent, render } from "@testing-library/react-native";
import React from "react";

import CreateButton from "./CreateButton";

describe("Given CreateButton component", () => {
  describe("When it's clicked", () => {
    test("Then it should call the freceives function", () => {
      const functionCreate = jest.fn();

      const screen = render(<CreateButton functionCreate={functionCreate} />);

      const button = screen.getByRole("button");

      fireEvent.press(button);
      expect(button).not.toBeNull();
      expect(functionCreate).toHaveBeenCalled();
    });
  });
});
