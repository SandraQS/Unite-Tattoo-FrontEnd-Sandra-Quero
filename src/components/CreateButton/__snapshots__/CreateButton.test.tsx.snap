// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given CreateButton component When it is rendered Then it should show match the last snapshot 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 20,
        "flex": 1,
        "flexDirection": "column",
        "margin": 5,
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "borderRadius": 20,
          "flex": 5,
          "height": 200,
          "justifyContent": "center",
        }
      }
    >
      <Image
        onError={[Function]}
        source={
          Object {
            "testUri": "../../../src/assets/crear-rombo.png",
          }
        }
        style={
          Array [
            Object {
              "height": 0,
              "width": 150,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
